### PR TITLE
Fix bug where BucketMapping.labels was incorrect when right=False

### PR DESCRIPTION
**Pull request by timvink**
This took some time to figure out, this small refactor also simplifies the code a bit more.

### CODE REVIEW

**Comment by sbjelogr** [comment]
Looks good!

File: tests/test_bucket_mapping.py [hunk 1]
```diff
@@ -88,13 +92,13 @@ def test_specials_numerical():
...
 
-    assert bucket.labels == {0: "(-inf, 3.0]", 1: "(3.0, 4.0]", 2: "(4.0, inf)", 3: "Missing", 4: "Special: =2"}
+    assert bucket.labels == {0: "(-inf, 3.0]", 1: "(3.0, 4.0]", 2: "(4.0, inf]", 3: "Missing", 4: "Special: =2"}
 
     # test that calling transform again does not change the labelling
     bucket.transform(x)
     bucket.transform(x)
 
-    assert bucket.labels == {0: "(-inf, 3.0]", 1: "(3.0, 4.0]", 2: "(4.0, inf)", 3: "Missing", 4: "Special: =2"}
+    assert bucket.labels == {0: "(-inf, 3.0]", 1: "(3.0, 4.0]", 2: "(4.0, inf]", 3: "Missing", 4: "Special: =2"}
```
Comment (sbjelogr) [inline tests/test_bucket_mapping.py:101]: Inclusive infinite does not make really sense, does it?

File: skorecard/bucket_mapping.py [hunk 2]
```diff
@@ -414,3 +386,33 @@ def __missing__(self, key):
...
+    if right:
+        b_left = "("
+        b_right = "]"
+    else:
+        b_left = "["
+        b_right = ")"
+
+    for i, boundary in enumerate(boundaries):
+        if i != len(boundaries) - 1:
+            labels[i] = f"{b_left}{boundary}, {boundaries[i+1]}{b_right}"
```
Comment (sbjelogr) [inline skorecard/bucket_mapping.py:416]: Noticed it in the test cases.

When the edge is `+/-inf`, mathematically speaking it makes no sense to add the inclusive bracket.
it should always be `(-inf,..`. and `...,inf)`

File: tests/test_bucket_mapping.py (hunk 1 above)
Comment (timvink) [reply tests/test_bucket_mapping.py:101]: It doesn't, but it made the implementation simpler. With this refactoring, improving the labelling is now changing one separate function. I suggest we do it later, separate PR

File: skorecard/bucket_mapping.py (hunk 2 above)
Comment (timvink) [reply skorecard/bucket_mapping.py:416]: Yup, separate PR.

**Pull request merged**